src/wp-includes/blocks.php

```diff
@@ -2420,6 +2421,48 @@ function build_query_vars_from_query_block( $block, $page ) {
 				}
 			}
 		}
+		if ( ! empty( $block->context['query']['format'] ) && is_array( $block->context['query']['format'] ) ) {
+			$formats   = $block->context['query']['format'];
+			$tax_query = array( 'relation' => 'OR' );
+
+			// The default post format, `standard`, is not stored in the database.
+			// If `standard` is part of the request, the query needs to exclude all post items that
+			// have a format assigned.
+			if ( in_array( 'standard', $formats, true ) ) {
+				$tax_query[] = array(
+					'taxonomy' => 'post_format',
+					'field'    => 'slug',
+					'terms'    => array(),
+					'operator' => 'NOT EXISTS',
+				);
+				// Remove the `standard` format, since it cannot be queried.
+				unset( $formats[ array_search( 'standard', $formats, true ) ] );
+			}
+			// Add any remaining formats to the tax query.
+			if ( ! empty( $formats ) ) {
+				// Add the `post-format-` prefix.
+				$terms = array_map(
+					static function ( $format ) {
+						return 'post-format-' . $format;
+					},
+					$formats
+				);
+				$tax_query[] = array(
+					'taxonomy' => 'post_format',
+					'field'    => 'slug',
+					'terms'    => $terms,
+					'operator' => 'IN',
+				);
+			}
+			if ( ! isset( $query['tax_query'] ) ) {
+				$query['tax_query'] = array();
+			}
+			// This condition is intended to prevent `$tax_query` from being added to `$query`
+			// if it only contains the relation.
```

Comment:
```suggestion
			/*
			 * This condition is intended to prevent `$tax_query` from being added to `$query`
			 * if it only contains the relation.
			 */
```